packages/beeq/src/components/tag/bq-tag.tsx

```diff
@@ -0,0 +1,233 @@
+import { Component, Element, Event, EventEmitter, h, Host, Method, Prop, State, Watch } from '@stencil/core';
+
+import { SIZE_TO_VALUE_MAP, TAG_SIZE, TAG_TYPE, TAG_VARIANT, TTagSize, TTagType, TTagVariant } from './bq-tag.types';
+import { hasSlotContent, validatePropValue } from '../../shared/utils';
+
+/**
+ * @part wrapper - The wrapper container `<div>` of the element inside the shadow DOM.
+ * @part prefix - The `<span>` tag element that acts as prefix container (when icon exists in front of tag).
+ * @part text - The `<div>` element containing the text of the tag component.
+ */
+@Component({
+  tag: 'bq-tag',
+  styleUrl: './scss/bq-tag.scss',
+  shadow: true,
+})
+export class BqTag {
+  // Own Properties
+  // ====================
+
+  private prefixElem: HTMLElement;
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqTagElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  @State() private hasPrefix = false;
+
+  // Public Property API
+  // ========================
+
+  /** The type of the tag component */
+  @Prop({ reflect: true }) size: TTagSize = 'medium';
+
+  /** If true, the tag component has color style */
+  @Prop({ reflect: true }) hasColor: boolean;
+
+  /** If true, the tag component can be removed */
+  @Prop({ reflect: true }) isRemovable: boolean;
```

Comment:
Can we use just `removable` instead? I think it makes it simpler, what do you think?

Reply:
Yes, sure thing.